Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3644,7 +3644,7 @@ void WebPageProxy::handleWheelEvent(const WebWheelEvent& wheelEvent)
     if (!hasRunningProcess())
         return;
 
-    if (drawingArea()->shouldSendWheelEventsToEventDispatcher()) {
+    if (drawingArea()->shouldSendWheelEventsToEventDispatcher() || m_hasActiveViewTransition) {
```

Comment:
Can we instead plumb this through like we do other `ReasonsForSynchronousScrolling` ?

Reply:
I'd still prefer this.

Reply:
I tried to address this in the commit message, `ReasonsForSynchronousScrolling` tries to wait (up to a limit), but doesn't stop the scrolls from being processed in the UI-side scroll tree representation.

The bug here is that the UI-side scroll tree isn't correctly (doesn't handle the layer reparenting, and with this change is just empty), so having the UI-side scroll tree handle the scroll events results in incorrect behaviour.

We need to entirely prevent them from reaching the scroll thread for it to work.